lib/TermUtil.ts

```diff
@@ -193,6 +193,51 @@ export function stringQuadToQuad<Q extends RDF.BaseQuad = RDF.Quad>(
   );
 }
 
+function escapeIRI(iriValue: string): string {
+  return iriValue.replace(escapeAll, replaceEscapedCharacter);
+}
+
+function escapeStringRDF(stringValue: string): string {
+  if (escape.test(stringValue)) {
+    stringValue = stringValue.replace(escapeAll, replaceEscapedCharacter);
+  }
+  return stringValue;
+}
+
+/* eslint-disable require-unicode-regexp */
```

Comment:
I guess we can just follow this rule, and add the `u` flag to the regexes?

Reply:
I think it clicked in my head all of a sudden. Adding the `u` flag means you have to be more careful matching on unicode code points, so it is not possible to match on only the first part of a surrogate pair. This means the regex has to inlcude both halves and thus become exactly like the second one. I'll commit it below.